### PR TITLE
fix: wrong speaker count

### LINKED ISSUE #6393: Wrong number of speakers
**Describe the bug**
https://eventyay.com/e/c0e175a3
Event has only 1 speaker but number of confirmed speakers is being shown as 5 on both frontend and Android.

Number of speakers issue

### PR DESCRIPTION
**Pull request by Tlazypanda**
Fixes #6393 

#### Short description of what this resolves:
Modified the speaker count method to fetch from Speaker model filtering out deleted speakers and joining with speaker session model in distinct manner to fetch the accurate state for the speaker as well.

#### Changes proposed in this pull request:

- Modified the speaker count method to fetch from Speaker model
- Filtering out deleted speakers 
- Join with speaker session link model to fetch accurate state count in distinct manner.

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by Tlazypanda** [comment]
@iamareebjamal can you kindly review? Thanks :v:

File: app/api/schema/event_statistics.py [hunk 1]
```diff
@@ -76,9 +76,18 @@ def sessions_canceled_count(self, obj):
...
-        ).count()
+        return (
+            Speaker.query.join(
+                SessionsSpeakersLink, SessionsSpeakersLink.speaker_id == Speaker.id
+            )
+            .distinct(Speaker.id)
+            .filter(
+                Speaker.event_id == obj.id,
+                Speaker.deleted_at == None,
+                SessionsSpeakersLink.session_state == state,
```
Comment (iamareebjamal) [inline app/api/schema/event_statistics.py:87]: Join session and use Session.state instead

Comment (Tlazypanda) [reply app/api/schema/event_statistics.py:87]: @iamareebjamal There is no column in Session for joining them should I add that in the schema? :sweat_smile:

Comment (iamareebjamal) [reply app/api/schema/event_statistics.py:87]: SessionSpeakerLink has session ID as well

Comment (Tlazypanda) [reply app/api/schema/event_statistics.py:87]: Ohh so you want me to join session in addition to session speaker link? what additional filtering do we need in that case @iamareebjamal

Comment (Tlazypanda) [reply app/api/schema/event_statistics.py:87]: I am getting a bit confused if we only link Session and SessionSpeakerLink we will lose our ability to filter out deleted speakers :sweat_smile:

Comment (iamareebjamal) [reply app/api/schema/event_statistics.py:87]: Please join the technical meeting at 7 PM and we'll discuss there. What is your email?

Comment (Tlazypanda) [reply app/api/schema/event_statistics.py:87]: itm2017004@iiita.ac.in @iamareebjamal

**Comment by Tlazypanda** [comment]
@iamareebjamal Updated mr :v: can you please review?

File: app/api/schema/event_statistics.py [hunk 2]
```diff
@@ -76,9 +76,19 @@ def sessions_canceled_count(self, obj):
...
 
     def speakers_count_type(self, obj, state='pending'):
-        return SessionsSpeakersLink.query.filter_by(
-            event_id=obj.id, session_state=state, deleted_at=None
-        ).count()
+        return (
+            Speaker.query.join(
+                SessionsSpeakersLink, SessionsSpeakersLink.speaker_id == Speaker.id
+            )
+            .join(Session, SessionsSpeakersLink.session_id == Session.id)
```
Comment (iamareebjamal) [inline app/api/schema/event_statistics.py:83]:
```suggestion
            Speaker.query.join(Speaker.sessions)
```

Comment (Tlazypanda) [reply app/api/schema/event_statistics.py:83]: Hey @iamareebjamal  I tried with this but it's not fetching it correctly then

Comment (iamareebjamal) [reply app/api/schema/event_statistics.py:83]: Strange. This query works for me

```python
Speaker.query.join(Speaker.sessions).distinct(Speaker.id).filter(Speaker.deleted_at == None, Session.state=='confirmed').count()
```

Comment (Tlazypanda) [reply app/api/schema/event_statistics.py]: @iamareebjamal ohh my bad had made a mistake now :sweat_smile:  have updated mr :v: Thanks for the suggestion!

**Comment by iamareebjamal** [comment]
Please check if SessionSpeakerLink.session_state is being used elsewhere. If not, please remove the column and run `python manage.py db migrate`

**Review by iamareebjamal — APPROVED** [review]
Great work! Thank You

**Comment by mariobehling** [comment]
Nice. Thank you!

**Pull request merged**